be/src/vec/exec/join/vhash_join_node.cpp

```diff
@@ -160,7 +161,7 @@ struct ProcessHashTableProbe {
             : _join_node(join_node),
               _batch_size(batch_size),
               _probe_rows(probe_rows),
-              _build_blocks(join_node->_build_blocks),
+              _build_blocks(join_node->_shared_hash_table_ctx._shared_structure->_build_blocks),
```

Comment:
I have a question for `process_data_in_hashtable` in this structure. Assume that for a right outer join, we have FI1 and FI2 share one hash table. If FI1 ended its probing before FI2, FI1 execute `process_data_in_hashtable` first. However, if a record (`record1`) doesn't match any record in FI1 but match some records in FI2, when FI1 execute `process_data_in_hashtable`, FI1 will assume `record1` doesn't match any and output NULL and record1 (line 549). After that, when FI2's probing continue, `record1` matches some records actually. So finally we got a wrong answer because FI1 output an unexpected row. Will this happen? Pls point out my mistake if I misunderstood this process. Thx

Reply:
you are right,so right join is prohibited in Fe.In the future, each shared thread can occupy a bit on the hash table to record whether the current row has been matched. If right join is used frequently, we can consider adding this feature